components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/core/ServerApiResourceManagementService.java

```diff
@@ -0,0 +1,473 @@
+                        ErrorMessage.ERROR_CODE_ADD_API_RESOURCE);
+            }
+        } catch (ApiResourceManagementException e) {
+            throw ApiResourceManagementEndpointUtil.handleApiResourceMgtException(e);
+        }
+        return buildApiResourceResponse(createdApiResource);
+    }
+
+    /**
+     * Build ApiResourceResponse from ApiResource.
+     *
+     * @param apiResource ApiResource object.
+     * @return ApiResourceResponse object.
+     */
+    private ApiResourceResponse buildApiResourceResponse(ApiResource apiResource) {
+
+        ApiResourceResponse apiResourceResponse = new ApiResourceResponse();
+        apiResourceResponse.setId(apiResource.getId());
+        apiResourceResponse.setName(apiResource.getName());
+        apiResourceResponse.setIdentifier(apiResource.getIdentifier());
+        apiResourceResponse.setDescription(apiResource.getDescription());
+        apiResourceResponse.setScopes(apiResource.getScopes().stream().map(this::buildScopeGetResponse)
+                .collect(Collectors.toList()));
+        apiResourceResponse.setRequiresAuthorization(apiResource.isRequiresAuthorization());
+        return apiResourceResponse;
+    }
+
+    /**
+     * Build ScopeGetModel from Scope.
+     *
+     * @param scope Scope object.
+     * @return ScopeGetModel object.
+     */
+    private ScopeGetModel buildScopeGetResponse(Scope scope) {
+
+        ScopeGetModel scopeGetModel = new ScopeGetModel();
+        scopeGetModel.setId(scope.getId());
+        scopeGetModel.setDisplayName(scope.getName());
+        scopeGetModel.setDescription(scope.getDescription());
+        return scopeGetModel;
+    }
+
+    /**
+     * Create Api resource from the Api resource creation model.
+     *
+     * @param apIResourceCreationModel Api resource creation model.
+     * @return Api resource.
+     */
+    private ApiResource createApiResource(ApiResourceCreationModel apIResourceCreationModel)
+            throws ApiResourceManagementException {
+
+        ApiResource apiResource = new ApiResource();
+        apiResource.setName(apIResourceCreationModel.getName());
+        apiResource.setIdentifier(apIResourceCreationModel.getIdentifier());
+        apiResource.setDescription(apIResourceCreationModel.getDescription());
+        apiResource.setScopes(createScopes(apIResourceCreationModel.getScopes()));
+        apiResource.setRequiresAuthorization(apIResourceCreationModel.getRequiresAuthorization());
+        ApiResourceManagementEndpointUtil.validateApiResource(apiResource);
+        return apiResource;
+    }
+
+    /**
+     * Create scopes from the scope creation models.
+     *
+     * @param scopeCreationModels Scope creation models.
+     * @return List of scopes.
+     */
+    private List<Scope> createScopes(List<ScopeCreationModel> scopeCreationModels)
+            throws ApiResourceManagementException {
+
+        List<Scope> scopes = new ArrayList<>();
+        if (scopeCreationModels == null) {
```

Comment:
Can't we use the isEmpty in Lists?

Reply:
scopeCreationModels can be null since, in the API creation payload it is not a mandatory field.